### PR TITLE
[KYUUBI #5457] [AUTHZ] Support RepairTable Commands for Hudi

### LINKED ISSUE #5457: [Subtask]  Support Repair table command
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues) and found no similar issues.

### Describe the subtask

 Support Repair table command

### Parent issue

#5285

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### _Why are the changes needed?_
To close #5457. Kyuubi authz support hudi repair table commands

- RepairHoodieTableCommand: https://github.com/apache/hudi/blob/master/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/RepairHoodieTableCommand.scala

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_
No

### CODE REVIEW

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/HudiCatalogRangerSparkExtensionSuite.scala [hunk 1]
```diff
@@ -127,6 +127,7 @@ class HudiCatalogRangerSparkExtensionSuite extends RangerSparkExtensionSuite {
       interceptContains[AccessControlException](
         doAs(someone, sql(s"ALTER TABLE $namespace1.$table1 ADD COLUMNS(age int)")))(
         s"does not have [alter] privilege on [$namespace1/$table1]")
+      sql("set hoodie.schema.on.read.enable=false")
```
Comment (AngersZhuuuu) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/HudiCatalogRangerSparkExtensionSuite.scala:130]: @yaooqinn After check it's caused by I forgot to set this parameter back.

Comment (yaooqinn) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/HudiCatalogRangerSparkExtensionSuite.scala:130]: Let's make sure to reach this line after L126 with a try-finally block?

Comment (AngersZhuuuu) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/HudiCatalogRangerSparkExtensionSuite.scala]: Done

**Review by yaooqinn — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**